tests/unit/conftest.py

```diff
@@ -115,16 +119,17 @@ def dbsession(connection):
     Adapted from https://docs.sqlalchemy.org/en/14/orm/session_transaction.html#joining-a-session-into-an-external-transaction-such-as-for-test-suites
     """
     transaction = connection.begin()
-    session = Session(autocommit=False, autoflush=False, bind=connection)
-    nested = connection.begin_nested()
+    session = ScopedSessionLocal()
+    nested = session.begin_nested()
 
     # If the application code calls session.commit, it will end the nested
     # transaction. Need to start a new one when that happens.
-    @event.listens_for(session, "after_transaction_end")
-    def end_savepoint(session, transaction):
+    @event.listens_for(ScopedSessionLocal, "after_transaction_end")
+    def end_savepoint(*args):
         nonlocal nested
-        if not nested.is_active:
-            nested = connection.begin_nested()
+        if nested.is_active:
```

Comment:
@leplatrem I'm looking at this again, and frankly I'm confused why the boolean flip here is still working.
From:
```
if not nested.is_active:
```
to
```
if nested.is_active:
```

This works, and makes more sense:
```python
    nested = session.begin_nested()

    @event.listens_for(ScopedSessionLocal, "after_transaction_end")
    def end_savepoint(_, trx):
        nonlocal nested
        if trx.nested and not nested.is_active:
            session.expire_all()
            nested = session.begin_nested()
```

But I'm not entirely sure what's happening here to make this work or not work.

Reply:
Ahh, I'm able to use the pattern straight from the docs
```python
def dbsession(connection):
    """Return a database session that rolls back.

    Adapted from https://docs.sqlalchemy.org/en/14/orm/session_transaction.html#joining-a-session-into-an-external-transaction-such-as-for-test-suites
    """
    transaction = connection.begin()
    session = ScopedSessionLocal()
    nested = connection.begin_nested()

    # If the application code calls session.commit, it will end the nested
    # transaction. Need to start a new one when that happens.
    @event.listens_for(ScopedSessionLocal, "after_transaction_end")
    def end_savepoint(*args):
        nonlocal nested
        if not nested.is_active:
            nested = connection.begin_nested()

    yield session
    session.close()
    transaction.rollback()
```

If I comment out [this line ](https://github.com/mozilla-it/ctms-api/blob/69750e77d3a7393af2721755e5187c3418555619/tests/unit/test_crud.py#L51)